### ISSUE TITLE
HTML lexer doesn't handle elements with dashes in tag name

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#553]
(Original issue 846 created by None on 2013-02-09T01:58:58.244769+00:00)

The html lexer generates ```<span class="err">```'s on closing elements with dashes in the name, e.g. You can see this in the following inline example that's using (presumably) the same HTML lexer:

```
#!html
<x-tag>Hello</x-tag>
```

Here's the the same code without specifying a lexer:

```
<x-tag>Hello</x-tag>
```

This isn't a standard HTML tag, but custom elements/web components will have dashes in their names (e.g, http://www.x-tags.org/index).

**Comment by Anteru** [comment]
(Original issue 846 last updated on 2013-05-19T08:24:40.484465+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by d6y on 2013-02-09T17:25:39.467696+00:00)

Possibly related, I have similar output for:

    <lift:notice_class>boxed</lift:notice_class>

Where by the _class part has different formatting, and the closing of the tag is not formatted:

```
#!HTML

<lift:notice_class>boxed</lift:notice_class>
```

**Comment by Anteru** [comment]
(Original comment by tshatch on 2013-05-10T01:09:21.522064+00:00)

I agree I've element names like those in the wild.  Richard, your should parse as xml in the meantime I think, since they're namespaced.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2013-05-19T08:24:40.476808+00:00)

Closes #846: support "strange" tag names (with ":", "_" or "-") in HTML lexer.

→ <<cset d466b2ef7c6c>>
